src/config/mod.rs

```diff
@@ -59,22 +59,31 @@ pub struct Config {
     pub selection_ui_auto_open: bool,
     pub use_virtual_text: UseVirtualText,
     pub echo_project_root: bool,
-    pub semantic_highlight_maps: HashMap<String, HashMap<String, String>>,
-    pub semantic_scope_separator: String,
     pub apply_completion_text_edits: bool,
     pub preferred_markup_kind: Option<Vec<MarkupKind>>,
     pub hide_virtual_texts_on_insert: bool,
     pub enable_extensions: Option<HashMap<String, bool>>,
     pub restart_on_crash: bool,
     pub max_restart_retries: u8,
+    /// semantic_token_mappings is a map of language id to a map of LSP semantic token type to vim
+    /// highlight group. This is used to fetch the correct vim highlight group for each semantic
+    /// token. The default State includes a generic mapping that can be used for any language id,
+    /// any other mappings included for specific language ids will be merged with the default one
+    /// before highlighting the document.
+    ///
+    /// For example:
+    ///
+    /// {
+    ///     "rust": { "type": "Type", "struct": "Structure" },
+    ///     "go": { "type": "Type", "struct": "Structure" },
+    /// }
+    pub semantic_token_mappings: HashMap<String, HashMap<String, String>>,
```

Comment:
I wonder if using the language as a key is the best approach, since:
1. There are multiple languages with the same LS (C and C++ are both clangd).
2. There are multiple LSs for the same language (for C++ there's clangd and ccls).

Maybe using the language server as a type somehow makes more sense? Or if they all use similar semantic token types just put them all in one map without dividing into languages?

Reply:
Yeah I did consider having them keyed by language server name, but considering that we have all other configs based on language id it felt more appropriate. Having said that, it does feel like having a single map for all of them might be a better idea, if users eventually want more flexibility and want to break it down by language server we can do so and support both anyways, but might as well keep it simple for now.